src/activity_item_handling.cpp

```diff
@@ -932,433 +939,1123 @@ static construction check_build_pre( const construction &con )
     return pre_con;
 }
 
-void activity_on_turn_blueprint_move( player_activity &, player &p )
+static bool character_has_skill_for( const player *p, const construction &con )
```

Comment:
```suggestion
static bool character_has_skill_for( const player &p, const construction &con )
```
No need for a pointer at all. Calling the function with a null pointer will just crash, so why even allow it?